### PR TITLE
Add date-time validation

### PR DESCRIPTION
**Pull request by Sabreen-Parveen**
- [ ] Contains functions which check date 
This PR is linked to #3

### CODE REVIEW

**Comment by dapryor** [comment]
Hey @Sabreen-Parveen

Thanks for the PR!

a few things:
1.  Since these are meant to be validations for some input, I think there should not be an optional required argument. It should always be required.

1. I have updated in the interface I am trying to maintain for validators in the issue for #3 .  If you could check that out and update you code, that would be appreciated.

1.  I would prefer to stick to functions only instead of classes.

1.  I think the more preferred approach here would be to make a function factory that allowed a user to specify a datetime string to validate against and if the user wanted to convert that to a datetime object, a post_processor could be used.

**Comment by Sabreen-Parveen** [comment]
Sure, I would modify my PR

File: question_framework/validation/validation.py [hunk 1]
```diff
@@ -57,3 +57,17 @@ def ip_range(x: str) -> bool:
         return False
     ip1, ip2 = split_ips
     return is_ip_check(ip1) and is_ip_check(ip2)
+
+def date_validation(x):
+  date_format = '%Y-%m-%d'
```
Comment (dapryor) [inline question_framework/validation/validation.py:62]: Hardcoding the date format limits the number of scenarios you can use this validation.

I think you should make this a function factory that takes a date format and creates/returns a function that utilizes that format to determine if the input is correctly formatted. The return function should take in a string and output a bool.

This function from the function factory should ONLY return  True or False. 

There are some examples of a function factory in the validation.py file.

File: question_framework/validation/validation.py [hunk 2]
```diff
@@ -57,3 +57,17 @@ def ip_range(x: str) -> bool:
...
+  try:
+    date_obj = datetime.strptime(x, date_format)
+    date_obj=str(date_obj).split()[0]
+    return(date_obj)
```
Comment (dapryor) [inline question_framework/validation/validation.py:66]: validation functions should only return True or False

File: question_framework/validation/validation.py [hunk 3]
```diff
@@ -57,3 +57,17 @@ def ip_range(x: str) -> bool:
...
+  except ValueError:
+    return "Invalid date format, it should be 'yyyy-mm-dd'"
```
Comment (dapryor) [inline question_framework/validation/validation.py:68]: validation functions should only return True or False

File: question_framework/validation/validation.py [hunk 4]
```diff
@@ -57,3 +57,17 @@ def ip_range(x: str) -> bool:
         return False
     ip1, ip2 = split_ips
     return is_ip_check(ip1) and is_ip_check(ip2)
+
+def date_validation(x):
+  date_format = '%Y-%m-%d'
+  try:
+    date_obj = datetime.strptime(x, date_format)
```
Comment (dapryor) [inline question_framework/validation/validation.py:64]: try checking out this stack overflow for ways to validate
https://stackoverflow.com/questions/16870663/how-do-i-validate-a-date-string-format-in-python/16870682

**Review by dapryor — CHANGES_REQUESTED** [review]

File: question_framework/validation/validation.py (hunk 4 above)
Comment (Sabreen-Parveen) [reply question_framework/validation/validation.py:64]: I will correct it.

**Comment by Sabreen-Parveen** [comment]
I have made changes. Please review my PR

**Comment by dapryor** [comment]
Hey @Sabreen-Parveen 
It seems your function still does not allow for dynamic datetime patterns to be used.  I would like to see a function factory similar to the x_hex_character_validation_gen.

Also you will need to rebase your code so we can merge it AND write a few test for your function

**Pull request closed without merging**